daffodil-core/src/main/scala/org/apache/daffodil/dpath/Expression.scala

```diff
@@ -1510,6 +1510,14 @@ case class FunctionCallExpression(functionQNameString: String, expressions: List
         FNOneArgExpr(functionQNameString, functionQName, args,
           NodeInfo.String, NodeInfo.AnyAtomic, FNLocalName1(_, _))
 
+      case (RefQName(_, "namespace-uri", FUNC), args) if args.length == 0 =>
+        FNZeroArgExpr(functionQNameString, functionQName,
+          NodeInfo.AnyURI, NodeInfo.AnyAtomic, FNNamespaceUri0(_, _))
+
+      case (RefQName(_, "namespace-uri", FUNC), args) if args.length == 1 =>
+        FNOneArgExpr(functionQNameString, functionQName, args,
+          NodeInfo.AnyURI, NodeInfo.AnyAtomic, FNNamespaceUri1(_, _))
```

Comment:
Also the argument type of NodeInfo.AnyAtomic feels wrong to me. This is what local-name uses, and I think that's wrong too. I think this should just be NodeInfo.Exists. That should ensure that a path to a node is passed in (and not a string/int/whatever) and that the node exist when run.

Reply:
I changed the argType to NodeInfo.Exists, although a test case still was able to pass an int to fn:namespace-uri anyway.